pkg/resource/nodegroup/options.go

```diff
@@ -213,7 +222,8 @@ func (o *NodegroupOptions) PreCreate() error {
 		o.Taints = append(o.Taints, Taint{Key: "nvidia.com/gpu", Effect: "NoSchedule"})
 	}
 
-	// AMI Lookup is currently only for Amazon Linux 2 EKS Optimized AMI and clusters that aren't fully private
+	// AMI Lookup is currently only for Amazon Linux 2 / Amazon Linux 2023 EKS Optimized AMI
+	// and clusters that aren't fully private
 	if o.OperatingSystem != "AmazonLinux2" || o.IsClusterPrivate {
```

Comment:
```suggestion
	if !strings.HasPrefix(o.OperatingSystem, "AmazonLinux") || o.IsClusterPrivate {
```
`eksdemo` started taking full ownership of AMI lookups for Amazon Linux 2 as [`eksctl` doesn't fully support Inf2 instances](https://github.com/eksctl-io/eksctl/issues/7683).

This condition has to be updated to include AMI lookup for Amazon Linux 2023 as returning here will skip the AMI lookup and let `eksctl` do it.